sphinx/domains/c.py

```diff
@@ -2801,13 +2798,16 @@ def parser():
             else:
                 break
         param = self._parse_parameters(paramMode)
-        if param is None and len(arrayOps) == 0:
+        if (
+            param is None and
+            len(arrayOps) == 0 and named and
             # perhaps a bit-field
-            if named and paramMode == 'type' and typed:
-                self.skip_ws()
-                if self.skip_string(':'):
-                    size = self._parse_constant_expression()
-                    return ASTDeclaratorNameBitField(declId=declId, size=size)
+            paramMode == 'type' and typed
+        ):
+            self.skip_ws()
+            if self.skip_string(':'):
+                size = self._parse_constant_expression()
+                return ASTDeclaratorNameBitField(declId=declId, size=size)
```

Comment:
Should not be changed, the original better conveys the semantics and is more readable. Here the transformation is even wrong because code has been moved across the comment!

Reply:
true